docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -123,221 +131,306 @@ You can initialize a selector with any of these objects.
 
-fixture `My fixture`
-    .page `http://www.example.com/`;
+#### withText
 
-const removeButton = Selector(cl => document.getElementsByClassName(cl), { text: 'Remove' });
-const secondButton = Selector(cl => document.getElementsByClassName(cl), { index: 1 });
+Method | Type | Description
+------ | ----- | -----
+`withText(text)` | Selector | Creates a selector that filters a matching set by the specified text.
+`withText(re)` | Selector | Creates a selector that filters a matching set using the specified regular expression.
 
-test('A shadowed remove button is focused', async t => {
-    expect(await removeButton('shadowed').focused).to.be.true;
-});
+#### filter
+
+Method | Type | Description
+------ | ----- | -----
+`filter(index)` | Selector | Creates a selector that filters a matching set by `index`.
+`filter(cssSelector)` | Selector | Creates a selector that filters a matching set by `cssSelector`.
+`filter(filterFn)` | Selector | Creates a selector that filters a matching set by `filterFn`; `filterFn` is a client function predicate that receives a node.
 
-test('The second disabled button is visible', async t => {
-    expect(await secondButton('disabled').visible).to.be.true;
+```js
+import { Selector } from 'testcafe';
+
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    const secondCheckBox = Selector('input[type=checkbox]').nth(1);
+    const checkedInputs  = Selector('input[type=checkbox]').filter(node => node.checked);
+    const windowsLabel   = Selector('label').withText('Windows');
+
+    await t
+        .click(secondCheckBox)
+        .expect(checkedInputs.count).eql(1)
+        .click(windowsLabel);
 });
 ```
 
-If both options are specified, nodes are first filtered by the `text` filter,
-then a node at the `index` position is selected from the remaining nodes.
-
 If all nodes are filtered out, the selector returns `null`.
 
-> The [text](selector-options.md#optionstext) and [index](selector-options.md#optionsindex) options
-> are also applied when the selector is called from another selector or a [client function](../obtaining-data-from-the-client.md).
+### Find Elements by DOM Hierarchy
 
-## Executing Selectors
+The selector API provides methods to find elements within a DOM hierarchy in jQuery style.
 
-To execute a selector, call it with the `await` keyword like you would do with regular async functions.
+#### find
+
+Property | Description
+------ | -----
+`find(cssSelector)` | Finds all descendants of all nodes in the matching set and filters them by `cssSelector`.
+`find(filterFn)` | Finds all descendants of all nodes in the matching set and filters them using `filterFn`; `filterFn` is a client function predicate that receives a node.
+
+#### parent
+
+Property | Description
+------ | -----
+`parent()` | Finds all parents of all nodes in the matching set.
```

Comment:
We need to give a note about ordering: first element in the set will be the closest parent. It's important, because snapshots and snapshot shorthands return values for the first element in the set.